keras/legacy/interfaces.py

```diff
@@ -172,3 +182,27 @@ def lstm_args_preprocessor(args, kwargs):
     value_conversions={'dim_ordering': {'tf': 'channels_last',
                                         'th': 'channels_first',
                                         'default': None}})
+
+legacy_upsampling1d_support = generate_legacy_interface(
+    allowed_positional_args=['size'],
+    conversions=[('length', 'size')])
+
+legacy_upsampling2d_support = generate_legacy_interface(
+    allowed_positional_args=['size', 'data_format'],
+    conversions=[('dim_ordering', 'data_format')],
+    value_conversions={'dim_ordering': {'tf': 'channels_last',
+                                        'th': 'channels_first',
+                                        'default': None},
+                       2: {'tf': 'channels_last',
```

Comment:
For simplicity, I would simply disallow `data_format` as a positional arg. It would be bad practice to pass it as positional anyway.